Week 5 - Cluedo, blue level - Nadia Mazzoni

Bluepoisson opened the pull request.

Bluepoisson committed:
Step 1 done.

And step 2.

Bluepoisson committed:
Blue level

All steps done.

Bluepoisson committed:
Images and styling

Updated image file and started styling the page.

Bluepoisson committed:
Moved file, font shadows

Blue level done.

Bluepoisson committed:
Updated Read Me

Read Me updated,
added margin-bottom to the cards for better stack view when mobile.

Bluepoisson committed:
Cleaned up code.

Cleaned up code and image file.

Bluepoisson committed:
Added reload.

rebkova commented on code/clue.js:
```diff
@@ -1,59 +1,220 @@
+  image: "images/weapon/rope-coil.svg",
+};
+
+const knife = {
+  name: "knife",
+  weight: 5,
+  material: "steel and wood",
+  image: "images/weapon/sacrificial-dagger.svg",
+};
```
Great job finding and adding images and more properties for weapons! I've also noticed you put in effort to find your own images of suspects :)

rebkova commented on code/clue.js:
```diff
@@ -1,59 +1,220 @@
+  document.getElementById("killerImage").src = `${mystery.killer.image}`;
+  document.getElementById("killerOccupation").innerHTML = `${mystery.killer.occupation}`;
+  document.getElementById("killerDescription").innerHTML = `${mystery.killer.description}`;
+  document.getElementById("killerCard").onclick = pickKiller;
```
Perhaps call the functions outside of their scopes (pickKiller, pickRoom, pickWeapon, revealMystery functions)?

rebkova commented on code/clue.js:
```diff
@@ -1,59 +1,220 @@
+  document.getElementById("mysteryBtn").onclick = revealMystery;
+  //Clears the revealMystery, need to setTimeOut
+  location.reload()
+};
```
The text after the mystery is revealed disappears very quickly. This is probably due to the fact that `location.reload()` lies inside of the function `revealMystery`, which upon being called, immediately reloads the page.
Perhaps try to move that part outside of the function and embed it into a button or something similar and see what happens?

rebkova commented on code/index.html:
```diff
@@ -4,34 +4,48 @@
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <link rel="icon" type="image/png" href="./images/clue2.jpg" />
```
Very nice with details like favicon :)

rebkova commented on code/style.css:
```diff
@@ -1,58 +1,145 @@
+  object-fit: cover;
+}
+
+.hero-text {
+  text-align: center;
+  position: absolute;
+  top: 5%;
+  left: 5%;
+  color: rgb(226, 223, 214);
+  text-shadow: 0 0 20px #fefcc9, 10px -10px 30px #feec85,
+    -20px -20px 40px #ffae34, 20px -40px 50px #ec760c, -20px -60px 60px #cd4606,
+    0 -80px 70px #973716, 10px -90px 80px #451b0e;
```
This shadow creates an amazing light effect on the text. Well done!

rebkova commented on code/style.css:
```diff
@@ -1,58 +1,145 @@
-body {
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+body,
+html {
+  height: 100%;
   margin: 0;
   font-family: "Courier New", Courier, monospace;
 }
```
If you use universal selector (*) with defined margin 0, do you still need to do that for the body and html tags?

rebkova commented on code/style.css:
```diff
@@ -1,58 +1,145 @@
+  background-color: transparent;
+}
+
+.hero-image {
+  height: 50%;
+  background-position: center;
+  background-repeat: no-repeat;
+  display: flex;
+  flex-wrap: wrap;
+  object-fit: cover;
+}
```
Great idea to have one large image as a background. It can be tricky with it causing the scrollbars, though. Perhaps try some of the tricks in the links below? 
https://css-tricks.com/perfect-full-page-background-image/  
https://www.w3schools.com/html/html_images_background.asp

rebkova commented on code/style.css:
```diff
@@ -1,58 +1,145 @@
+
+/* MEDIA QUERIES */
+
+@media screen and (min-width: 768px) {
+  .hero-text {
+    font-size: 20px;
+  }
+}
```
Great job making your page responsive! Double check the borders for the mobile and tablet (I assume the ones with font-size 20px and 25 px, respectively), you're almost there! :)

rebkova commented on code/style.css:
```diff
@@ -1,58 +1,145 @@
+
+/* MEDIA QUERIES */
+
+@media screen and (min-width: 768px) {
+  .hero-text {
+    font-size: 20px;
+  }
+}
+
+@media screen and (max-width: 768px) {
+  .hero-text {
+    font-size: 25px;
+    margin-top: 0;
+  }
+}
+
+@media screen and (min-width: 992px) {
+  .hero-text {
+    font-size: 40px;
+    margin: 50px;
+  }
+}
```
Congrats Nadia on finishing one weekly project more! 
You've created three clickable decks of cards that randomly generate a card and you sucessfully incorporated a clickable button that reveals a mystery hidden in those cards. In addition, you styled your page very nicely and thought of responsiveness and browser compatibility. Your code is clear to understand, indentations without issues, you used semantic tags properly.  
Considering you were still recovering you managed to meet all of the blue level requirements and submit your project on time. I am very proud of you and so should be you. :)

rebkova commented on code/index.html:
```diff
@@ -4,34 +4,48 @@
-    <h1 id="mystery"></h1>
-
-    <button>REVEAL THE CRIME</button>
+    <div class="hero-image">
+      <img
+        src="./images/piano.jpg"
+        alt="close up on old, dusty, dirty, beaten piano keys"
+      />
+      <p class="hero-text">PLAY CLUE, KLICK EACH CARD</p>
+    </div>
+    <main>
+      <section class="cards">
+        <article class="killer-deck" id="killerCard" onclick="pickKiller()">
+          <p>KILLER</p>
+          <p id="killerName"></p>
+          <p id="killerAge"></p>
+          <p id="killerOccupation"></p>
+          <img id="killerImage" height="300" />
+          <p id="killerDescription"></p>
+        </article>
+        <article class="weapon-deck" id="weaponCard" onclick="pickWeapon()">
+          <p>WEAPON</p>
+          <p id="weaponName"></p>
+          <p id="weaponWeight"></p>
+          <p id="weaponMaterial"></p>
+          <img id="weaponImage" height="250" />
+        </article>
+        <article class="room-deck" id="roomCard" onclick="pickRoom()">
+          <p>ROOM</p>
+          <p id="roomName"></p>
+        </article>
+      </section>
```
Very nice job with structuring and indentation.

Bluepoisson committed:
Fixed reload function

Updated the reload function to page reload with setTimeout.

HIPPIEKICK commented on code/clue.js:
```diff
@@ -1,59 +1,218 @@
+  document.getElementById("mystery").innerHTML = `The murder was comitted by ${mystery.killer.firstName} in the ${mystery.room} with a ${mystery.weapon.name}`;
+  document.getElementById("mysteryBtn").onclick = revealMystery;
+  //Function that reloads page 10s after revalMystery button is clicked
+  setTimeout(() => window.location.reload(), 10000);
```
Nice touch!

HIPPIEKICK commented on code/clue.js:
```diff
@@ -1,59 +1,218 @@
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// STEP 4 
+const revealMystery = () => {
+  document.getElementById("mystery").innerHTML = `The murder was comitted by ${mystery.killer.firstName} in the ${mystery.room} with a ${mystery.weapon.name}`;
+  document.getElementById("mysteryBtn").onclick = revealMystery;
```
What happens if you click the button and not all cards are picked? You could fix that by adding some conditional statements to this function to show one message if not all cards are picked, and only show this "The murder was committed..."-message after all cards are picked :)

HIPPIEKICK commented on code/index.html:
```diff
@@ -4,34 +4,48 @@
+          <p id="weaponName"></p>
+          <p id="weaponWeight"></p>
+          <p id="weaponMaterial"></p>
+          <img id="weaponImage" height="250" />
```
Keep the styling in the CSS ;)

HIPPIEKICK reviewed (changes requested):
@Bluepoisson Please fix the little bug so that it doesn't reveal the mystery until all cards are picked. Otherwise, great job!

@rebkova Keep it up with the constructive comments 👍

Bluepoisson committed:
Added mystery validation as per request.

Added if statement to the revealMystery function to ensure all cards are picked in order to display the mystery.
